cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractTwillProgramController.java

```diff
@@ -119,7 +119,7 @@ protected final void doStop() throws Exception {
     Futures.getUnchecked(twillController.terminate());
   }
 
-  protected final TwillController getTwillController() {
+  public final TwillController getTwillController() {
```

Comment:
this is usually a sign that some abstraction is being broken and that code is in the wrong place. If somebody gets the twill controller here and messes with it, it will cause this class to malfunction.